File: haskell/0013.hs
ans s [] = s
ans (s:ss) (0:xs) = s:(ans ss xs)
ans s (n:xs) = ans (n:s) xs

main = do
  c <- getContents
  let i = map read $ lines c :: [Int]
      o = ans [] i
  mapM_ print o

File: rust/0013.rs
use std::io::prelude::*;
use std::str::FromStr;

fn ans(d:i32,s:&mut Vec<i32>) {
    if d == 0 {
        let o = s.pop();
        println!("{}",o.unwrap());
    } else {
        s.push(d);
    }
}

fn main() {
    let stdin = std::io::stdin();
    let mut s : Vec<i32> = Vec::new();
    for line in stdin.lock().lines() {
        let d = i32::from_str(&line.unwrap()).unwrap();
        ans(d,&mut s);
    }
}
